firstvoices/backend/management/commands/copy_site.py

```diff
@@ -0,0 +1,532 @@
+        # To circumvent checks added to media models to prevent modification of original file
+        audio_file._state.adding = True
+        audio_file.save()
+
+        audio_file.speakers.set(updated_speakers)
+
+    # Copy over rest of the people who are not attached as speakers to any audio
+    person_list = Person.objects.filter(site=source_site, audio_set__isnull=True)
+    for person in person_list:
+        person.id = uuid.uuid4()
+        person.site = target_site
+        person.save()
+
+
+def copy_images(source_site, target_site, image_map):
+    image_files = Image.objects.filter(site=source_site)
+    for image_file in image_files:
+        # Content
+        target_file = ImageFile(
+            content=UploadedFile(image_file.original.content.file),
+            site=target_site,
+            import_job=None,
+        )
+        target_file.save()
+        image_file.original = target_file
+
+        image_file.thumbnail = None
+        image_file.small = None
+        image_file.medium = None
+        image_file.site = target_site
+
+        source_img_id = image_file.id
+        target_img_id = uuid.uuid4()
+        image_map[source_img_id] = target_img_id
+
+        image_file.id = target_img_id
+        image_file._state.adding = True
+        image_file.save()
+
+
+def copy_videos(source_site, target_site, video_map):
+    video_files = Video.objects.filter(site=source_site)
+    for video_file in video_files:
+        # Content
+        target_file = VideoFile(
+            content=UploadedFile(video_file.original.content.file),
+            site=target_site,
+            import_job=None,
+        )
+        target_file.save()
+        video_file.original = target_file
+
+        video_file.thumbnail = None
+        video_file.small = None
+        video_file.medium = None
+        video_file.site = target_site
+
+        source_video_id = video_file.id
+        target_video_id = uuid.uuid4()
+        video_map[source_video_id] = target_video_id
+
+        video_file.id = target_video_id
+        video_file._state.adding = True
+        video_file.save()
+
+
+def copy_galleries(source_site, target_site, image_map):
+    galleries = Gallery.objects.filter(site=source_site)
+    for gallery in galleries:
+        gallery_items = list(gallery.galleryitem_set.all())
+
+        gallery.site = target_site
+
+        target_cover_img_id = image_map[gallery.cover_image.id]
+        gallery.cover_image = Image.objects.get(id=target_cover_img_id)
+
+        gallery.id = uuid.uuid4()
+        gallery.save()
+
+        updated_gallery_items = []
+        for gallery_item in gallery_items:
+            new_gallery_item = GalleryItem(
+                gallery=gallery,
+                image=Image.objects.get(id=image_map[gallery_item.image.id]),
+                ordering=gallery_item.ordering,
+            )
+            new_gallery_item.save()
+            updated_gallery_items.append(new_gallery_item)
```

Comment:
If you're already updating all of these to eliminate extra db lookups, you could also refactor all the operations that update the m2m relations to use a common method. Something like:

    def update_related(instance, related_field_name, related_id_map):
        related_manager = getattr(instance, related_field_name)
        target_related = [related_id_map[s.id] for s in related_manager.all()]
        related_manager.set(target_related)
        instance.save # optional, might make sense to leave this out